modules/eks/locals.tf

```
@@ -1,3 +1,18 @@
 locals {
   availability_zone_ids = toset([for subnet in data.aws_subnet.this : subnet.availability_zone_id])
-}
+  # Mapping from user-friendly AMI type names to actual AMI prefixes
+  ami_type_prefix_map = {
+    "alinux2"    = "amazon-eks-node"
+    "alinux2023" = "amazon-eks-node-al2023-x86_64-standard"
```

It's better to avoid to set architecture in the prefix. Maybe also arm64 should be used
```suggestion
    "alinux2023" = "amazon-eks-node-al2023"
```